### PR TITLE
Dockerized Server Deployment

### PR DESCRIPTION
**Pull request by rHorsey**

### CODE REVIEW

**Comment by rHorsey** [comment]
@nllong Can you start a code review on this? I'm test fixes and one spreadsheet integration bug away from claiming this is completed.

**Comment by rHorsey** [comment]
@nllong @brianlball FYI I think this is now complete. I'm building an AMI 2.0.1-rc0 from this that updates to the 2.0.1 Ubuntu build that @macumber got us a few weeks back. The dockerfile has been updated and passes on CI!

File: Dockerfile [hunk 1]
```diff
@@ -118,11 +120,11 @@ RUN apt-get update \
     && curl -SLO $OPENSTUDIO_DOWNLOAD_URL \
     && gdebi -n $OPENSTUDIO_DOWNLOAD_FILENAME \
     && rm -f $OPENSTUDIO_DOWNLOAD_FILENAME \
-    && rm -rf /usr/local/lib/openstudio-$OPENSTUDIO_VERSION/ruby/2.0/openstudio/sketchup_plugin \
+    && rm -rf /usr/SketchUpPlugin \
     && rm -rf /var/lib/apt/lists/*
 
 # Add RUBYLIB link for openstudio.rb
-ENV RUBYLIB /usr/local/lib/site_ruby/2.0.0
+ENV RUBYLIB /usr/Ruby
```
Comment (nllong) [inline Dockerfile]: this is where openstudio.rb lives now?

File: Dockerfile [hunk 2]
```diff
@@ -174,7 +176,7 @@ RUN bundle exec rake assets:precompile
 ADD /server /opt/openstudio/server
 ADD .rubocop.yml /opt/openstudio/.rubocop.yml
 # Run bundle again, because if the user has a local Gemfile.lock it will have been overriden
-RUN bundle install
+RUN bundle update
```
Comment (nllong) [inline Dockerfile]: Are we sure we want to run update? I prefer bundle install so that it doesn't try to update any gems that we don't have locked down.

**Comment by nllong** [comment]
looks good, only a few comments...

File: Dockerfile (hunk 2 above)
Comment (rHorsey) [inline Dockerfile]: Great catch. This change was derived thanks to the following build error:
```shell
Using rake 11.3.0
Using i18n 0.8.0
Using json 1.8.6
Using minitest 5.10.1
Using thread_safe 0.3.5
Using builder 3.2.3
Using erubis 2.7.0
Using mini_portile2 2.1.0
Using rack 1.6.5
Using mime-types-data 3.2016.0521
Using arel 6.0.4
Using minitar 0.6
Your Gemfile.lock is corrupt. The following gem is missing from the DEPENDENCIES
section: 'minitar-cli'
```
Is there a better way of fixing this?

File: Dockerfile (hunk 1 above)
Comment (rHorsey) [inline Dockerfile]: Yep. It's a whole new world!

File: Dockerfile (hunk 2 above)
Comment (nllong) [inline Dockerfile]: just do `bundle install`

Comment (nllong) [inline Dockerfile]: oh and remove the lock file.

Comment (rHorsey) [inline Dockerfile]: Will do. Tks!

Comment (rHorsey) [inline Dockerfile]: Resolved. See https://github.com/NREL/OpenStudio-server/pull/130/commits/9063eb2ad8731c0552c509061230019f25273891

**Pull request merged**